spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala

```diff
@@ -16,12 +16,116 @@
 
 package com.intel.analytics.bigdl.nn
 
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.RandomGenerator.RNG
+
+/**
+ * DataFormat describe the dimension of weights
+ */
+sealed trait DataFormat
+
 /**
- * Initialization method to initialize bias and weight
+ * OutputFirst indicating that output is in the lower dimension
+ * of the weight. Take 2 dimensional weight for example, the first
+ * dimension is the output, and the second dimension is the input.
  */
-sealed trait InitializationMethod
+case object OutputFirst extends DataFormat
+
+/**
+ * InputFirst indicating that input is in the lower dimension
+ * of the weight. Take 2 dimensional weight for example, the first
+ * dimension is the input, and the second dimension is the output.
+ */
+case object InputFirst extends DataFormat
+
 
-case object Default extends InitializationMethod
+
+/**
+ * Initialization method to initialize bias and weight.
+ * The init method will be called in Module.reset()
+ */
+
+trait InitializationMethod {
+
+  type Shape = Array[Int]
+
+  /**
+    * Initialize the given weight and bias.
+    *
+    * @param weight    the weight to initialize
+    * @param bias      the bias to initialize, do nothing if bias is None
+    * @param dataFormat       the data format of weight indicating the dimension order of
+    *                  the weight. "output_first" means output is in the lower dimension
+    *                  "input_first" means input is in the lower dimension.
+    */
+  def init[T](weight: Tensor[T], bias: Option[Tensor[T]], dataFormat: DataFormat = OutputFirst)
```

Comment:
somes layer with weight and bias should accept two initMethod. some layers with weight should accept one initMethod and with bias only should accept one initMethod